### ISSUE TITLE
Cast call results in "Ambigous method call"

### ISSUE THREAD

**Issue by disarray2077** [beefytech/Beef#1403]
The below code fails to compile because of the following error:

```
ERROR: Ambiguous method call
  while compiling 'test6.Program'
  > 'test6.Program.Test.operator System.Collections.List<int>(test6.Program.Test self)' is a candidate at line 223:27 in d:\BeefLang\test6\src\Program.bf
    public static implicit operator List<int>(Self self)
                           ^^^^^^^^
  > 'test6.Program.Test.operator System.Span<int>(test6.Program.Test self)' is a candidate at line 226:27 in d:\BeefLang\test6\src\Program.bf
    public static implicit operator Span<int>(Self self)
                           ^^^^^^^^
```

I guess this is happening because both List<int> and Span<int> are IEnumerable, but I don't think this should happen since:
- There's no way to disambiguate the cast call.
- The Test struct itself is a IEnumerable, so Beef shouldn't try to cast it.

Anyway, here is the code:
```
static
{
	[Inline]
	public static void ForEach<TList, TValueType, TDlg>(this TList list, TDlg dlg)
		where TList : concrete, IEnumerable<TValueType>
		where TDlg : delegate void(TValueType)
	{
		for (let item in list)
			dlg(item);
	}
}

namespace test6
{
	class Program
	{
		struct Test : IEnumerable<int>
		{
			public static implicit operator List<int>(Self self)
				=> null;

			public static implicit operator Span<int>(Self self)
				=> null;

			public IEnumerator<int> GetEnumerator()
				=> default;
		}

		public static int Main(String[] args)
		{
			Test test = .();
			test.ForEach((i) => i);

			return 0;
		}
	}
}
```

**Comment by disarray2077** [comment]
I just found another issue in the code of this issue, if you right click `ForEach` in the `test.ForEach((i) => i);` line, the IDE crashes.

**Comment by bfiete** [comment]
Fixed at 1ee32434a5293ec1bd11ca5d9a1cabafcfd779ca

I couldn't reproduce the crash even with the last nightly build, though. Maybe post a crash dump?

**Comment by disarray2077** [comment]
I also had trouble reproducing it this time, probably it actually isn't related to this issue at all but anyway...
Crash dump:
[crash1.zip](https://github.com/beefytech/Beef/files/7982418/crash1.zip)

**Comment by bfiete** [comment]
Ah, good. There were two bugs-
1) The lambda capture phase would incorrect suggest fixits for lambda parameters (since the params don't exist yet in that pass) as "Create field 'i' in 'Program'".
2) Fixits for adding fields on types with comptime-emitted code would incorrect search for an insertion point for the field within the emitted code buffer rather than the type definition, results in an OOB index

#1 fixed at 6c3944170d5bef3c7ec7db889ce586a2d69a05bf (along with some other things)
#2 fixed at 438394099a05a82f52d7cd8508a59116a91f700b
